File: user-service/src/main/java/socialapp/userservice/common/mapper/AddressMapper.java
package socialapp.userservice.common.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import socialapp.userservice.model.dto.AddressDto;
import socialapp.userservice.model.entity.Address;

@Mapper
public interface AddressMapper {

    Address toModel(AddressDto addressDto);

    void update(@MappingTarget Address address,AddressDto addressDto);
}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/common/mapper/AddressMapper.kt
package socialapp.ktuserservice.common.mapper

import org.mapstruct.Mapper
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers
import socialapp.ktuserservice.model.dto.AddressDto
import socialapp.ktuserservice.model.entity.Address

@Mapper(componentModel = "spring")
interface AddressMapper {

    companion object{
        val INSTANCE = Mappers.getMapper(AddressMapper::class.java)
    }

    fun toModel(addressDto: AddressDto): Address

    fun update(address: Address, @MappingTarget addressDto: AddressDto)
}
